### PR TITLE
[14.0] [mig] pos_supplierinfo_search

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

**Comment by dsolanki-initos** [comment]
@flotho It would be great if you could add the reviews here.

**Review by fshah-initos — APPROVED** [review]
Looks Good To Me.
Functional Tests and code review.

**Review by brendapaniagua — COMMENTED** [review]
Good work! but It seems that it is not possible to search products by supplier name.

**Comment by chienandalu** [comment]
> Good work! but It seems that it is not possible to search products by supplier name.

Was it possible before?

**Comment by dsolanki-initos** [comment]
> Good work! but It seems that it is not possible to search products by supplier name.

IT is possible to search you need to set the vendor for product to get it work.

**Comment by dsolanki-initos** [comment]
@chienandalu can you please review and merge the PR?

**Comment by dsolanki-initos** [comment]
@chienandalu can you please review and merge the PR?

**Comment by dsolanki-initos** [comment]
It would be great if we could get the reviews here

**Comment by dsolanki-initos** [comment]
It would be great if we could get the reviews here

**Comment by dsolanki-initos** [comment]
It would be great if we could get the reviews here

**Comment by dsolanki-initos** [comment]
Hi @eLBati It would be nice if you can add reviews

File: pos_supplierinfo_search/models/product.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from odoo import fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    # technical field used in POS frontend
+    supplier_data_json = fields.Char(
+        "Supplier data dict", readonly=True, compute="_compute_supplier_data_json"
+    )
```
Comment (ivantodorovich) [inline pos_supplierinfo_search/models/product.py]:
```suggestion
    supplier_data_json = fields.Char(
        "Supplier data dict",
        help="Technical field: Used in POS frontend to search products by supplierinfo",
        compute="_compute_supplier_data_json",
    )
```

- Readonly is already True by default for computed fields
- Move comment to field help

File: pos_supplierinfo_search/models/product.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            res = []
+            for s in t.seller_ids:
+                res.append(
+                    {
+                        "supplier_name": s.name.display_name,
+                        "supplier_product_code": s.product_code or "",
+                        "supplier_product_name": s.product_name or "",
+                    }
+                )
+            t.supplier_data_json = json.dumps(res)
```
Comment (ivantodorovich) [inline pos_supplierinfo_search/models/product.py]:
```suggestion
        for rec in self:
            rec.supplier_data_json = json.dumps(
                [
                    {
                        "supplier_name": s.name.display_name,
                        "supplier_product_code": s.product_code or "",
                        "supplier_product_name": s.product_name or "",
                    }
                    for s in rec.seller_ids
                ]
            )
```

Not blocking, but you could rewrite it like this for readability

**Review by ivantodorovich — APPROVED** [review]
Minor comments

**Pull request merged**